### ISSUE TITLE
Synchronize issue: offline edits to removed objects

### ISSUE THREAD

**Issue by appy-one** [appy-one/acebase-client#3]
When using AceBaseClient with a local cache db, the following can occur:

* client1 makes changes to an object while offline
* client2 removes the object while online
* client1 goes online, pushes its changes to the server 
* A "broken" object is now stored in the database

Example:

Consider a contact in the db: ```{ name: 'Ewout', email: 'me@appy.one', phone: '+31612345678', last_called: (date) }```

client1 updates ```last_called``` while offline:
```js
db.ref('contacts/ewout').update({ last_called: new Date() });
```
client2 deletes the contact while online:
```js
db.ref('contacts/ewout').remove();
```
client1 goes online and pushes the change to the server. causing the stored contact to be "broken": ```{ last_called: (date) }``` because it was re-created by client1's update.

This should be prevented somehow.

**Comment by appy-one** [comment]
I'm currently testing moving schema validation from ```acebase-server``` to ```acebase```, so it can be done at storage level. This could prevent those broken nodes to be created, while also making schema validation possible for standalone databases without a server endpoint. Schema validation would then also become available for IndexedDB databases in the browser.

**Comment by appy-one** [comment]
Schema validation is now available in `acebase` v1.3.0
Next step is to update `acebase-server` to use the schema validation in `acebase` and remove its own validation code.

**Comment by appy-one** [comment]
`acebase-server` now uses `acebase`'s schema validation. The "broken object" issue can now be prevented by setting schema rules.
